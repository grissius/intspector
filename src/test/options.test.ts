import { getOptions, setOptions } from '..';

describe('Options', () => {
    test('getOptions', () => {
        expect(getOptions()).toMatchSnapshot();
    });
    test('setOptions', () => {
        const opts = setOptions({ strict: false });
        expect(opts).toMatchObject(getOptions());
        expect(opts).toMatchSnapshot();
    });
    test('setOptions (ignore)', () => {
        const opts = setOptions({ baseUrl: 'foo' }, true);
        expect(opts).toMatchObject(getOptions());
        expect(opts).toMatchSnapshot();
    });
});
