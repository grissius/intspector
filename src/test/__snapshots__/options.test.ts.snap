// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options getOptions 1`] = `
Object {
  "baseUrl": "src",
  "configFilePath": undefined,
  "declaration": true,
  "esModuleInterop": true,
  "lib": Array [
    "lib.es2017.d.ts",
  ],
  "module": 1,
  "moduleResolution": 2,
  "outDir": "dist",
  "rootDir": "src",
  "sourceMap": true,
  "strict": true,
  "target": 1,
}
`;

exports[`Options setOptions (ignore) 1`] = `
Object {
  "baseUrl": "foo",
}
`;

exports[`Options setOptions 1`] = `
Object {
  "baseUrl": "src",
  "configFilePath": undefined,
  "declaration": true,
  "esModuleInterop": true,
  "lib": Array [
    "lib.es2017.d.ts",
  ],
  "module": 1,
  "moduleResolution": 2,
  "outDir": "dist",
  "rootDir": "src",
  "sourceMap": true,
  "strict": false,
  "target": 1,
}
`;
